src/utils/inactiveUsers.js

```diff
@@ -0,0 +1,113 @@
+const Promise = require('bluebird');
+const { LockAcquisitionError } = require('ioredis-lock');
+
+const { getInternalData } = require('./userData');
+const removeUser = require('./removeUser');
+
+const { USERS_ACTIVATE,
+  USERS_ACTIVATE_AUDIENCE,
+} = require('../constants');
+
+const lockKey = `${USERS_ACTIVATE}:data-lock`;
+
+// Get user ids having score lt current time - interval
+async function getInactiveUsers(redis, interval) {
+  const start = '-inf';
+  const end = Date.now() - (interval * 1000);
+  return redis.zrangebyscore(USERS_ACTIVATE, start, end);
+}
+
+async function getAudience(redis, userId) {
+  const audience = await redis.hget(USERS_ACTIVATE_AUDIENCE, userId);
+  return audience;
+}
+
+async function deleteUser(userId, audience) {
+  const { redis } = this;
+  const context = { redis, audience };
+
+  const internal = await Promise
+    .bind(context, userId)
+    .then(getInternalData);
+
+  const transaction = redis.multi();
+
+  removeUser(transaction, internal, audience);
+  return transaction.exec();
+}
+
+/**
+ * Add user id to inacive users list
+ * @param {ioredis} redis
+ * @param {userId} userId
+ * @param {audience[]} audience
+ */
+function addInactiveUser(redis, userId, audience) {
+  const created = Date.now();
+  redis.zadd(USERS_ACTIVATE, userId, created);
+  redis.hset(USERS_ACTIVATE_AUDIENCE, userId, JSON.stringify(audience));
+}
+
+/**
+ * Remove user id from inactive users list
+ * @param {ioredis} redis
+ * @param {userId} userId
+ */
+function removeInactiveUser(redis, userId) {
+  redis.zrem(USERS_ACTIVATE, userId);
+  redis.hdel(USERS_ACTIVATE_AUDIENCE, userId);
+}
+
+/**
+ * Clean all users, who did't pass activation
+ * Call in `service` context
+ */
+async function cleanInactiveUsers() {
+  const { redis } = this;
+  const { deleteInactiveAccounts } = this.config;
+
+  const lock = await this.dlock.once(lockKey);
+  const inactiveAccounts = await getInactiveUsers(redis, deleteInactiveAccounts);
+
+  inactiveAccounts.forEach(async (acc) => {
+    const accAudiences = getAudience(redis, acc);
+    await deleteUser.bind(this, acc, accAudiences);
+    await removeInactiveUser(redis, acc);
+  });
+
+  lock.release().reflect();
+}
+
+/**
+ * Returns inactive users cleanup task
+ * @param {Mfleet} service
+ * @returns {NodeJS.Timeout}
+ */
+function getCleanerTask(service) {
+  const { deleteInactiveAccountsInterval: interval } = service.config;
```

Comment:
multiple instances would conflict - we want to scale this up easily and, even though, lock would generally prevent things from breaking - I'd suggest to avoid them if possible